Makefile

```diff
@@ -161,16 +164,45 @@ build-dev: ## Build the project in development mode (Supports `ENVIRONMENT=true`
 
 build-all: build-x86_64-unknown-linux-musl build-armv7-unknown-linux-musleabihf build-aarch64-unknown-linux-musl ## Build the project in release mode for all supported platforms
 
-build-x86_64-unknown-linux-gnu: ## Build dynamically linked binary in release mode for the x86_64 architecture
-	$(RUN) build-x86_64-unknown-linux-gnu
+build-x86_64-unknown-linux-gnu: ## Build dynamically linked binary in release mode for the x86_64 architecture. (Docker only)
+	$(CONTAINER_TOOL) buildx build \
+		--platform linux/amd64 \
+		--build-arg BASEIMAGE="centos:7" \
+		--build-arg BOOTSTRAPPER="scripts/environment/bootstraps/centos-7.sh" \
+		--file scripts/environment/Dockerfile \
+		--tag $(ENVIRONMENT_UPSTREAM):x86_64-unknown-linux-gnu \
+		.
+	$(CONTAINER_TOOL) run \
```

Comment:
I plan to abstract all this to generic musl/gnu builders next PR in this series, then the PR after will abstract over platforms.